pkg/route/registry/route/rest.go

```diff
@@ -103,6 +110,13 @@ func (rs *REST) Update(ctx kapi.Context, obj runtime.Object) (<-chan apiserver.R
 	if errs := validation.ValidateRoute(route); len(errs) > 0 {
 		return nil, errors.NewInvalid("route", route.Name, errs)
 	}
+
+	err := rs.loadCerts(route.TLS)
```

Comment:
If we are following the same assumptions we had on v2 then the REST API should always expect the data and the cli takes a file path as input, read its data and send it to the api. Flags that take a file path should always have the `-file` suffix. Please confirm @smarterclayton.

That said, it's ok to just force people to set the data for now, we can adjust it later.